diff --git i/hyprctl/src/main.cpp w/hyprctl/src/main.cpp
index 0a33f3ed..09ddc6fe 100644
--- i/hyprctl/src/main.cpp
+++ w/hyprctl/src/main.cpp
@@ -271,7 +271,7 @@ int requestIPC(std::string_view filename, std::string_view arg) {
     sockaddr_un serverAddress = {0};
     serverAddress.sun_family  = AF_UNIX;
 
-    std::string socketPath = getRuntimeDir() + "/" + instanceSignature + "/" + filename;
+    std::string socketPath = getRuntimeDir() + "/" + std::string(instanceSignature) + "/" + std::string(filename);
 
     strncpy(serverAddress.sun_path, socketPath.c_str(), sizeof(serverAddress.sun_path) - 1);
 
diff --git i/src/helpers/Monitor.cpp w/src/helpers/Monitor.cpp
index 3be5be40..9d106bd1 100644
--- i/src/helpers/Monitor.cpp
+++ w/src/helpers/Monitor.cpp
@@ -693,7 +693,7 @@ bool CMonitor::applyMonitorRule(SMonitorRule* pMonitorRule, bool force) {
         std::ranges::sort(sortedModes, sortFunc);
         if (sortedModes.size() > 3)
             sortedModes.erase(sortedModes.begin() + 3, sortedModes.end());
-        requestedModes.insert_range(requestedModes.end(), sortedModes | std::views::reverse);
+        requestedModes.insert(requestedModes.end(), sortedModes.rbegin(), sortedModes.rend());
     };
 
     // last fallback is always preferred mode
diff --git i/start/src/helpers/Nix.cpp w/start/src/helpers/Nix.cpp
index da66183e..678a1f85 100644
--- i/start/src/helpers/Nix.cpp
+++ w/start/src/helpers/Nix.cpp
@@ -9,9 +9,6 @@
 #include <hyprutils/string/String.hpp>
 #include <hyprutils/os/Process.hpp>
 #include <hyprutils/os/File.hpp>
-
-#include <glaze/glaze.hpp>
-
 using namespace Hyprutils::String;
 using namespace Hyprutils::OS;
 
@@ -95,14 +92,7 @@ bool Nix::shouldUseNixGL() {
         return false;
     }
 
-    auto json = glz::read_json<glz::generic>(proc.stdOut());
-    if (!json) {
-        g_logger->log(Hyprutils::CLI::LOG_ERR, "failed to obtain hyprland version string (bad json)");
-        return false;
-    }
-
-    const auto FLAGS  = (*json)["flags"].get_array();
-    const bool IS_NIX = std::ranges::any_of(FLAGS, [](const auto& e) { return e.get_string() == std::string_view{"nix"}; });
+    const bool IS_NIX = proc.stdOut().find("\"nix\"") != std::string::npos;
 
     if (IS_NIX) {
         const auto NAME = getFromEtcOsRelease("NAME");
